src/streamlink/plugin/plugin.py

```diff
@@ -398,6 +400,15 @@ def get_streams(self, *args, **kwargs):
     def _get_streams(self):
         raise NotImplementedError
 
+    def get_title(self):
```

Comment:
I think these should return `None` instead, and the `create_title` method should handle the `None` response. This is better because if a Plugin cannot find the title or category for some stream, then it doesn't have to look up what the default is and return it - easier for Plugins to not screw it up ;)

Reply:
Oe maybe it could be one method that returns a `dict` so that plugins could add their own metadata, like `episode` or `season`... but, maybe that is crazy.

Reply:
I kind of like the `dict` idea, but it does create the problem of "secret format codes" for users. If there are only 3 (4 counting the category/game synonym) format codes, then it's easier for users to know all the possible format codes without reading documentation for each plugin. Plus I think it's easier to write for plugin developers...

Reply:
Yeah, probably. 

My only concern is that `{game}` is very specific and `{category}`/`{author}` don't apply to everything. If we include `{game}`, why not include more? if we don't include more options, why include `{game}`?

Reply:
To me it seems like a lot of online streaming is either gaming oriented like twitch, or more like youtube. So that's the perspective I had adding those format codes. `category` doesn't have to be implemented, and I think every stream should have a human/corporate_entity that starts it (`author`)

I'm open to including more format codes, but I don't know what they would be. I'd need the perspective of someone more involved with streamlink. You said maybe `episode` and `season`, but I think that could acceptably be covered by `category`, or really, that particular information would usually be in the `title` of the stream.

Reply:
yeah, if I were implementing `{title} {category} {author}` for a plugin where the `season` and `episode` are available, I'd put it in `{title}` like `S01E01 It all begins`

Reply:
There are a few gaming plugins, but the majority are live TV - which I guess doesn't really have `{game}` or `{category}` :) You are right about `{author}`, that should be applicable in most cases.